src/or/connection.c

```diff
@@ -2604,25 +2621,37 @@ connection_read_proxy_handshake(connection_t *conn)
   return ret;
 }
 
+struct replacement_s
```

Comment:
We need to document this type.  What's it for?  What do the members mean?

Reply:
Also let's give this a typedef name, and use that instead: we tend not to use "struct foo" in Tor.

Reply:
Fixed in ee3fbf1195052019c73c8c5e43b64bf5f17ab4a2 and 88de87e5d90ddaeaac21e8910428f6c3c9f68e98 .